src/dart/parse.rs

```diff
@@ -217,6 +255,8 @@ impl<'a> Parser<'a> {
         result.ok()
     }
 
+    /// Returns an identifier if the current token is an identifier
+    /// except reserved words, otherwise returns an error.
```

Comment:
"an identifier if the current token is an identifier" phrasing is redundant, also, it cares about keywords, not reserved words, AFAIR.
Maybe "Parses a non-keyword identifier, otherwise returns an error."?